File: Exercism/c/difference-of-squares/src/difference_of_squares.c
#include <math.h>
#include "difference_of_squares.h"

long square_of_sum(int n) {
    long sum = 0;

    for (int i = 0 ; i <= n ; ++i)
        sum += i;

    return pow(sum, 2);
}

long sum_of_squares(int n) {
    long sum = 0;

    for (int i = 0 ; i <= n ; ++i)
        sum += pow(i, 2);

    return sum;
}

long difference_of_squares(int n) {
    return square_of_sum(n) - sum_of_squares(n);
}

File: Exercism/dart/difference-of-squares/lib/difference_of_squares.dart
import 'dart:math';

class DifferenceOfSquares {
  int squareOfSum(int x) {
    int sum = 0;

    for (int i = 1 ; i <= x ; i++) {
      sum += i;
    }

    return pow(sum, 2) as int;
  }

  int sumOfSquares(int x) {
    int sum = 0;

    for (int i = 1 ; i <= x ; i++) {
      sum += pow(i, 2) as int;
    }

    return sum;
  }

  int differenceOfSquares(int x) {
    return squareOfSum(x) - sumOfSquares(x);
  }
}
